contracts/croncat-agents/src/contract.rs

```diff
@@ -689,3 +691,34 @@ fn on_task_created(
     let response = Response::new().add_attribute("action", "on_task_created");
     Ok(response)
 }
+fn on_task_completed(
+    deps: DepsMut,
+    info: MessageInfo,
+    args: AgentOnTaskCompleted,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.may_load(deps.storage)?.unwrap();
+    AGENTS
+        .may_load(deps.storage, &args.agent_id)?
+        .ok_or(ContractError::AgentNotRegistered {})?;
+    croncat_manager_contract::assert_caller_is_manager_contract(
+        &deps.querier,
+        &config,
+        &info.sender,
+    )?;
+    let mut stats = AGENT_STATS
+        .may_load(deps.storage, &args.agent_id)?
+        .unwrap_or_default();
```

Comment:
```suggestion
    let mut stats = AGENT_STATS
        .load(deps.storage, &args.agent_id)?;
```
I don't like it being default here. It should just load, since we checked this agent is active already on `proxy_call`, and, in case, if it didn't load we better fail the transaction, instead of giving false agent stats progression!